extensions/mongo-client/deployment/src/main/java/io/quarkus/mongo/deployment/MongoClientProcessor.java

```diff
@@ -0,0 +1,73 @@
+package io.quarkus.mongo.deployment;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.bson.codecs.configuration.CodecProvider;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+
+import com.mongodb.client.MongoClient;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.builditem.SslNativeConfigBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+import io.quarkus.mongo.ReactiveMongoClient;
+import io.quarkus.mongo.runtime.MongoClientConfig;
+import io.quarkus.mongo.runtime.MongoClientProducer;
+import io.quarkus.mongo.runtime.MongoClientTemplate;
+import io.quarkus.runtime.RuntimeValue;
+
+public class MongoClientProcessor {
+
+    @BuildStep
+    AdditionalBeanBuildItem registerClientProducerBean() {
+        return AdditionalBeanBuildItem.unremovableOf(MongoClientProducer.class);
+    }
+
+    @BuildStep
+    CodecProviderBuildItem collectCodecProviders(CombinedIndexBuildItem indexBuildItem) {
+        Collection<ClassInfo> codecProviderClasses = indexBuildItem.getIndex()
+                .getAllKnownImplementors(DotName.createSimple(CodecProvider.class.getName()));
+        List<String> names = codecProviderClasses.stream().map(ClassInfo::toString).collect(Collectors.toList());
```

Comment:
Not excited about using the `toString()`. People sometimes change that without thinking that it could be used.